doc/rest-authentication.md

```diff
@@ -0,0 +1,106 @@
+# Glusterd2 REST API authentication
+
+Glusterd2 REST API authentication is based on [JWT](https://jwt.io). If REST
+authentication is enabled then each client request should include
+`Authorization` header as example below.
+
+    Authorization: bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnbHVzdGVyY2xpIiwiaWF0IjoxNTMxNzI5NzQyLCJleHAiOjE1MzE3Mjk3NTJ9._WIwO7PrHUHIT62SdzfkyNjqD1GEgX2cYqN8ACZCtaw
+
+**Note**: REST Authentication can be disabled by adding `restauth=false`
+in Glusterd2 config file(Default path is
+`/var/lib/glusterd2/glusterd2.toml` in case of rpm installation)
+
+## Default user
+
+When `glusterd2` starts for the first time, it creates the
+`$GLUSTERD2_STATEDIR/auth` file which will contain the secret. If
```

Comment:
Do we document anywhere what GLUSTERD2_STATEDIR path mean? Might be good to point out to a link.

Reply:
> Do we document anywhere what GLUSTERD2_STATEDIR path mean? Might be good to point out to a link
I don't think we have a document about the paths yet. We should do it.

Apart from that, I have a concern about using the GD2 secret by external tools.

The secret is meant to be accessible only by GD2. GD2 should be using this secret to create and handout tokens. Having the clients access the secret is a big security risk. Later in the document, we ask that the secret be copied out externally and have external client generate their own token. This is a serious lapse of security. Because the secret can be passed along to unintended users and those users would have full access to the GD2 cluster.

Ideally, as I mentioned earlier GD2 should be the one generating and handing out tokens. Even for `glustercli` it would be preferrable that you use token. GD2 should generate a token for CLI and save it to a known place. And we should add a way to generate tokens to GD2 (ReST or CLI to be decided).

Sorry about bringing this up now. I should have been paying attention when the actual PR was being reviewed. I'll open up a new issue about `glustercli` not using the secret file directly. We can merge this PR without waiting for that, but only if it is modified to address my comments later on.